src/psyclone/tests/domain/gocean/transformations/gocean_extract_test.py

```diff
@@ -434,179 +446,10 @@ def test_driver_generation_flag(tmpdir, create_driver):
-    # names.
-    driver = tmpdir.join("driver-psy_extract_example_with_various_variable_"
-                         "access_patterns-invoke_1_compute_kernel:compute_"
-                         "kernel_code:r0.f90")
-    assert driver.isfile()
-
-    with driver.open("r") as driver_file:
-        driver_code = driver_file.read()
-
-    expected = """
-      REAL(KIND=8), allocatable, dimension(:,:) :: out_fld_post
-      REAL(KIND=8), allocatable, dimension(:,:) :: out_fld_post0
-      REAL(KIND=8), allocatable, dimension(:,:) :: out_fld
-      REAL(KIND=8), allocatable, dimension(:,:) :: in_out_fld_post0
-      REAL(KIND=8), allocatable, dimension(:,:) :: in_out_fld
-      CALL extract_psy_data%ReadVariable("in_out_fld", in_out_fld)
-      CALL extract_psy_data%ReadVariable("in_out_fld_post0", in_out_fld_post0)
-      CALL extract_psy_data%ReadVariable("out_fld_post0", out_fld_post0)
-      ALLOCATE (out_fld, mold=out_fld_post0)
-      CALL extract_psy_data%ReadVariable("out_fld_post", out_fld_post)"""
-
-    ordered_lines_in_text(expected.split("\n"), driver_code)
-
-    # Now test that more than one variable clash is handled. The third
-    # invoke uses:
-    # "out_fld" as output field
-    # "out_fld_post" as input field (first clash --> suffix becomes "_post0")
-    # "out_fld_post0" as input+output field (next clash --> suffix = "_post1")
-    psy, invoke = get_invoke("driver_test.f90",
-                             GOCEAN_API, idx=2, dist_mem=False)
-    schedule = invoke.schedule
-    # We don't check the driver, we already tested that the
-    # driver picks up the adjusted suffix above
-    etrans.apply(schedule.children[0])
-    extract_code = str(psy.gen)
-
-    # Check that *out_fld* is declared correctly: it is only declared as
-    # output value, so must use key out_fld_post1 once, and not be declared
-    # as input value:
-    assert 'PreDeclareVariable("out_fld_post1", out_fld)' in extract_code
-    assert 'PreDeclareVariable("out_fld", out_fld)' not in extract_code
-
-    # Check that *out_fld_post* (input/output) is declared correctly. It must
-    # be declared twice: once for the input value using the original variable
-    # name, and once as output using the "_post1" suffix"
-    assert 'PreDeclareVariable("out_fld_post", out_fld_post)' in extract_code
-    assert 'PreDeclareVariable("out_fld_post_post1", out_fld_post)' \
-        in extract_code
-
-    # Check that *out_fld_post0* is declared correctly: as input-only
-    # variable it must be declared once for using the original variable name.
-    assert 'PreDeclareVariable("out_fld_post0", out_fld_post0)' in extract_code
-    assert 'PreDeclareVariable("out_fld_post0_post1", out_fld_post0)' \
-        not in extract_code
-
-
 # -----------------------------------------------------------------------------
 def test_driver_loop_variables(tmpdir):
     '''Test that loop variables are not stored. ATM this test
```

Comment:
The test is x-failing ... explicit;y further down. Reason for this is that this test failed earlier due to some other, unrelated bug ... which went undetected because it was an x-failing test. This now makes sure that the beginning also works as expected before triggering the x-fail.

Reply:
Oh yes, I've been bitten by that too.